commerce/domain/inventory/cart/calculator/classes/InventoryCartCalculatorSampleTest.cls

```diff
@@ -0,0 +1,199 @@
+        Id cartId = arrangeCartWithSpecifiedStatus(cartStatus);
+        arrangeThreeCartItems(cartId);
+        return CartExtension.CartTestUtil.getCart(cartId);
+    }
+    
+    private static ID arrangeCartWithSpecifiedStatus(CartExtension.CartStatusEnum cartStatus) {
+        Account account = new Account(Name = ACCOUNT_NAME);
+        insert account;
+
+        WebStore webStore = new WebStore(Name = WEBSTORE_NAME, OptionsCartCalculateEnabled = true);
+        insert webStore;
+
+        WebCart webCart = new WebCart(
+                Name = CART_NAME,
+                WebStoreId = webStore.Id,
+                AccountId = account.Id,
+                Status = cartStatus.name());
+        insert webCart;
+        return webCart.Id;
+    }
+
+    private static List<ID> arrangeThreeCartItems(ID cartId) {
+        CartDeliveryGroup deliveryGroup = new CartDeliveryGroup(Name = DELIVERYGROUP_NAME, CartId = cartId);
+        insert deliveryGroup;
+
+        CartItem cartItem1 = new CartItem(
+                Name = CART_ITEM1_NAME,
+                CartId = cartId,
+                CartDeliveryGroupId = deliveryGroup.Id,
+                Quantity = 3,
+                SKU = SKU1_NAME,
+                Type = CartExtension.SalesItemTypeEnum.PRODUCT.name());
+        insert cartItem1;
+
+        CartItem cartItem2 = new CartItem(
+                Name = CART_ITEM2_NAME,
+                CartId = cartId,
+                CartDeliveryGroupId = deliveryGroup.Id,
+                Quantity = 3,
+                SKU = SKU2_NAME,
+                Type = CartExtension.SalesItemTypeEnum.PRODUCT.name());
+        insert cartItem2;
+
+        CartItem cartItem3 = new CartItem(
+                Name = CART_ITEM3_NAME,
+                CartId = cartId,
+                CartDeliveryGroupId = deliveryGroup.Id,
+                Quantity = 3,
+                SKU = SKU3_NAME,
+                Type = CartExtension.SalesItemTypeEnum.PRODUCT.name());
+        insert cartItem3;
+        return new List<ID>{cartItem1.Id, cartItem2.Id, cartItem3.Id};
+    }
+
+    private static void assertNoCartValidationOutputs(CartExtension.Cart cart, Integer expectedCVOCount) {
+        String errorString = '';
+        Iterator<CartExtension.CartValidationOutput> cvoIterator = cart.getCartValidationOutputs().iterator();
+        while (cvoIterator.hasNext()) {
+            errorString += cvoIterator.next().getMessage() + '; ';
+        }
+        Assert.areEqual(expectedCVOCount, cart.getCartValidationOutputs().size(), 'No CartValidationOutputs expected, but was: ' + errorString);
+    }
+    
+    private static void addCvoToCart( CartExtension.Cart cart) {
+         CartExtension.CartItemList cartItemCollection = cart.getCartItems();
+            Iterator<CartExtension.CartItem> cartItemCollectionIterator = cartItemCollection.iterator();
+    
+        	Integer i = 0;
+            while (cartItemCollectionIterator.hasNext()) {
```

Comment:
Why do we need to iterate over cartItems to add just one CartValidationOutput? Looks like this can be simplified to 
```
CartExtension.CartValidationOutput cvo = new CartExtension.CartValidationOutput(CartExtension.CartValidationOutputTypeEnum.INVENTORY, CartExtension.CartValidationOutputLevelEnum.ERROR);
cart.getCartValidationOutputs().add(cvo);
```